server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java

```diff
@@ -702,16 +701,11 @@ public void run() {
     try {
 
       final AtomicReference<Throwable> err = new AtomicReference<>();
-      final AtomicLong timeSinceLastCompletion = new AtomicLong(0L);
+      final AtomicBoolean isIdle = new AtomicBoolean(true);
 
       while (!shutdown) {
 
-        idleProcessCheck(() -> {
-          return timeSinceLastCompletion.get() == 0
-              /* Never started a compaction */ || (timeSinceLastCompletion.get() > 0
-                  && (System.nanoTime() - timeSinceLastCompletion.get())
-                      > idleReportingPeriodNanos);
-        });
+        idleProcessCheck(isIdle::get);
```

Comment:
```suggestion
        idleProcessCheck(()->true);
```

Reply:
I also tried this with no luck. With this approach I do not see the value for the compactor idle metric leave the value its is initialized to (-1).

Reply:
I would add logging to `idleProcessCheck` the logs all the variables and run the test and see what that looks like each time its called.

Reply:
I figured out what was going on. In order to have the metric be set to idle, two consecutive calls to `AbstractServer.idleProcessCheck()` need to be made passing in true. Depending on if I used the `idleProcessCheck(isIdle::get);` or `idleProcessCheck(()->true);`, the metric value would get stuck since the compactor code loops while a compaction is happening and the value was not updated. By adding `idleProcessCheck(() -> true);` in the main loop, we ensure the metric is correctly marked idle when not compacting. The nested compaction loop handles setting the metric to busy and back to idle after compaction completes.